apache/incubator-kie-tools
# kie-issues#1827: Refactor Management Console to support multiple runtimes and OpenID Connect

Closes https://github.com/apache/incubator-kie-issues/issues/1827

- Adds support to connect to multiple runtimes at the same time;
- Support for authenticated runtimes (using OpenID Connect);
- Refactor/rewrite of enveloped components, using and/or fixing React Hooks;
- Support for bookmarkable-urls.

`nonce` and `prompt` are commented out: https://github.com/apache/incubator-kie-issues/issues/1830

---

## Demo

https://github.com/user-attachments/assets/474d8ce9-25a3-4f9f-99a5-6da38a79ae42 


## How to test:

- Build everything with `pnpm -r -F runtime-tools-management-console-webapp... build:dev`;
- Run the webapp and the services (2 quarkus apps + Keycloak) with `pnpm -F runtime-tools-management-console-webapp start:all`;
- Connect the runtimes to the Management Console (see video above);
- Create processes:
  - Secured runtime:
    - ```
      export SECURED_RUNTIME_IDP_TOKEN=$(\
          curl -X POST http://localhost:9071/realms/management-console-dev-webapp-realm/protocol/openid-connect/token \
          -H 'content-type: application/x-www-form-urlencoded' \
          -d "client_id=management-console-dev-webapp" \
          -d "grant_type=password" \
          -d "username=jdoe" \
          -d "password=jdoe" \
          -d "scope=openid" | jq --raw-output '.access_token' \
       )
      ```
    - ```
      curl -H "Content-Type: application/json" -H "Accept: application/json" -X POST http://localhost:8080/hiring -H "Authorization: Bearer $SECURED_RUNTIME_IDP_TOKEN" -d '{"candidateData": { "name": "Jon", "lastName": "Snow", "email": "jon@snow.org", "experience": 5, "skills": ["Java", "Kogito", "Fencing"]}}'
      ```
  - Unsecured runtime:
    - ```
      curl -H "Content-Type: application/json" -H "Accept: application/json" -X POST http://localhost:8081/hiring -d '{"candidateData": { "name": "Jon", "lastName": "Snow", "email": "jon@snow.org", "experience": 5, "skills": ["Java", "Kogito", "Fencing"]}}'
      ```
